### PR TITLE
Support connection limti configuration for roles

### PR DESCRIPTION
**Pull request by vin01**

### CODE REVIEW

File: init.sls [hunk 1]
```diff
@@ -134,6 +134,15 @@ createuser-{{ index }}:
     {% endif %}
     - user: postgres
 
+{% set connection_limit = config.get('connection_limit') %}
+{% if connection_limit %}
+set_user_{{ config['username'] }}_connection_limit:
+  cmd.run:
+    - name: psql -U postgres -c "ALTER ROLE {{ config['username'] }} CONNECTION LIMIT {{ connection_limit }};"
+    - unless: psql -U postgres -c "SELECT rolconnlimit FROM pg_roles WHERE rolname = '{{ config['username'] }}';" | grep -w {{ connection_limit }}
```
Comment (chr4) [inline init.sls]: Using `-t` should be less error prone than grepping with `-w`?

Comment (chr4) [reply init.sls]: Also I think `grep -q` is a good idea

**Review by chr4 — CHANGES_REQUESTED** [review]
Thanks for the contribution! I think this one is very useful.

File: tests/integration/postgresql/postgresql_spec.rb [hunk 2]
```diff
@@ -138,6 +138,11 @@
     its('output') { should match /example_role/ }
   end
 
+  # Assert that connection limit is reflected correctly
+    describe sql.query("SELECT rolconnlimit FROM pg_roles WHERE rolname='example_role'") do
+    its('output') { should eq('100') }
+  end
```
Comment (chr4) [inline tests/integration/postgresql/postgresql_spec.rb:144]: Can you add a test that a user that's not in the pillar will stay at `-1`? I think this is a good check to ensure upcoming changes might not mess with all users, etc.

File: init.sls (hunk 1 above)
Comment (vin01) [reply init.sls]: `-t` does not seem to exist, perhaps you meant a different flag? this is an integer field, `-w` would match the exact value

File: tests/integration/postgresql/postgresql_spec.rb (hunk 2 above)
Comment (vin01) [reply tests/integration/postgresql/postgresql_spec.rb:144]: good idea, updated tests 👍

File: init.sls (hunk 1 above)
Comment (chr4) [reply init.sls]: I meant `-t` for `psql`. But using `-w` sounds great then, as it prevents false positives.

Comment (vin01) [reply init.sls]: ah, got it. just updated 👍

**Pull request merged**